fvm/meter/weighted/meter.go

```diff
@@ -75,6 +197,14 @@ func WithComputationWeights(weights ExecutionEffortWeights) WeightedMeterOptions
 
 // WithMemoryWeights sets the weights for the memory intensities
 func WithMemoryWeights(weights ExecutionMemoryWeights) WeightedMeterOptions {
+	if int(common.MemoryKindLast)-len(weights) != 1 {
+		return func(m *Meter) {
+			for kind, weight := range weights {
+				m.memoryWeights[kind] = weight
+			}
+		}
+	}
+
 	return func(m *Meter) {
 		m.memoryWeights = weights
 	}
```

Comment:
I found this confusing despite knowing what was going on. Might just be be but I think this is clearer:

```suggestion
	// Replace all weights
	if int(common.MemoryKindLast)-len(weights) == 1 {
		return func(m *Meter) {
			m.memoryWeights = weights
		}
	}

	// Replace just some weights
	return func(m *Meter) {
		for kind, weight := range weights {
			m.memoryWeights[kind] = weight
		}
	}	
```